estate/models/estate_property.py

```diff
@@ -0,0 +1,63 @@
+from odoo import fields, models
+from dateutil.relativedelta import relativedelta
+
+class EstateProperty(models.Model):
+    _name='estate.property'
+    _description='Estate Property'
+
+    name=fields.Char(string='Title', required=True)
+    description=fields.Text()
+    postcode=fields.Char()
+    date_availablity=fields.Date(string='Available From', copy=False, default=fields.Date.today()+ relativedelta(months=3))
+    expected_price=fields.Float(required=True)
+    selling_price=fields.Float(readonly=True, copy=False)
+    bedrooms=fields.Integer(default=2)
+    living_area=fields.Integer(string='Living Area (sqm)')
+    facades=fields.Integer()
+    garage=fields.Boolean()
+    garden=fields.Boolean()
+    garden_area=fields.Integer(string='Garden Area (sqm)')
+    garden_orientation=fields.Selection(
+        selection=[
+            ('north','North'),
+            ('south','South'),
+            ('east','East'),
+            ('west','West'),
+        ]
+    )
+    active=fields.Boolean(default=True)
+    state=fields.Selection(
+        required=True,
+        default='new',
+        copy=False,
+        selection=[
+            ('new','New'),
+            ('offer_received','Offer Received'),
+            ('offer_accepted','Offer Accepted'),
+            ('sold','Sold'),
+            ('cancelled','Cancelled'),
+        ],
+    )
+    property_type_id=fields.Many2one(
+        comodel_name='estate.property.type',
+        string='Property Type',
+    )
+    salesman=fields.Many2one(
+        comodel_name='res.users',
+        string='Salesman',
+        default=lambda self: self.env.user,
+    )
+    buyer=fields.Many2one(
+        comodel_name='res.partner',
+        string='Buyer',
+        copy=False,
+    )
+    tag_ids=fields.Many2many(
+        comodel_name='estate.property.tag',
+        string='Tags',
+    )
+    offer_ids=fields.One2many(
+        comodel_name="estate.property.offer",
+        inverse_name='property_id',
```

Comment:
```suggestion
        inverse_name='property_id',
```
what is the use case of inverse_name and what happens if it is not defined?

Reply:
inverse_name is required in One2many fields to map the relationship to a corresponding Many2one field in the target model.

Without it, Odoo can't determine how records in the target model relate to each other, causing a no inverse field error. 

This ensures proper bidirectional access and data integrity.

Reply:
resolve it